app/src/main/java/com/wireguard/android/fragment/AddTunnelsSheet.kt

```diff
@@ -0,0 +1,115 @@
+        }
+
+        override fun onStateChanged(bottomSheet: View, newState: Int) {
+            if (newState == BottomSheetBehavior.STATE_COLLAPSED) {
+                dismiss()
+            }
+        }
+    }
+
+    override fun getTheme(): Int {
+        return R.style.BottomSheetDialogTheme
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        if (savedInstanceState != null) dismiss()
+        return inflater.inflate(R.layout.add_tunnels_bottom_sheet, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        view.viewTreeObserver.addOnGlobalLayoutListener(object :
+                ViewTreeObserver.OnGlobalLayoutListener {
+            override fun onGlobalLayout() {
+                view.viewTreeObserver.removeOnGlobalLayoutListener(this)
+                val dialog = dialog as BottomSheetDialog? ?: return
+                val bottomSheet: FrameLayout = dialog.findViewById(materialR.id.design_bottom_sheet)
+                        ?: return
+                behavior = BottomSheetBehavior.from(bottomSheet)
+                behavior.state = BottomSheetBehavior.STATE_EXPANDED
+                behavior.peekHeight = 0
+                behavior.addBottomSheetCallback(bottomSheetCallback)
+                dialog.findViewById<MaterialButton>(R.id.create_empty)?.setOnClickListener {
+                    dismiss()
+                    onRequestCreateConfig()
+                }
+                dialog.findViewById<MaterialButton>(R.id.create_from_file)?.setOnClickListener {
+                    dismiss()
+                    onRequestImportConfig()
+                }
+                dialog.findViewById<MaterialButton>(R.id.create_from_qrcode)?.setOnClickListener {
+                    dismiss()
+                    onRequestScanQRCode()
+                }
+            }
+        })
+        val gradientDrawable = GradientDrawable().apply {
+            setColor(requireContext().resolveAttribute(R.attr.colorBackground))
+        }
+        view.background = gradientDrawable
+    }
+
+    override fun dismiss() {
+        super.dismiss()
+        behavior.removeBottomSheetCallback(bottomSheetCallback)
+    }
+
+    private fun requireTargetFragment(): Fragment {
+        return requireNotNull(targetFragment) { "A target fragment should always be set" }
+    }
+
+    private fun onRequestCreateConfig() {
+        startActivity(Intent(activity, TunnelCreatorActivity::class.java))
+    }
+
+    private fun onRequestImportConfig() {
+        val intent = Intent(Intent.ACTION_GET_CONTENT).apply {
+            addCategory(Intent.CATEGORY_OPENABLE)
+            type = "*/*"
+        }
+        requireTargetFragment().startActivityForResult(
+                Intent.createChooser(intent, "Choose ZIP or conf"),
```

Comment:
Please use the same string we had earlier from somewhere. "ZIP or conf" is definitely not something real.

Reply:
Ack